compiler/optimizer/VPHandlers.cpp

```diff
@@ -9786,8 +9831,9 @@ static TR::Node *constrainIfcmpeqne(OMR::ValuePropagation *vp, TR::Node *node, b
           //of a call and the type of a receiver (e.g. abstract, interface, normal classes)
           //if findSingleImplementer returns a method than it must be the implementation Inliner used since
           //it would be the only one available
-          if (objectClass && rhsClass && callClass && !lhs->isFixedClass() &&
-              vp->comp()->fe()->isInstanceOf (rhsClass, objectClass, true, true, true) == TR_yes &&
+          if (objectClass && callClass && !isFixedClass &&
+              (((isMethodTest && inlinedMethodClass && vp->comp()->fe()->isInstanceOf (objectClass, inlinedMethodClass, true, true, true) == TR_yes))
```

Comment:
We should pass `false` as the last parameter to `isInstanceOf`, otherwise we may miss a AOT validation record for the relationship of the two classes. We should fix it for the existing one as well.